File: mac/Marquee/API/Models/JobsModels.swift
import Foundation

// Settings → Jobs (api-v1.md §13).

extension API {
    /// A scheduled maintenance job ("Run now" → `jobs.run(id)`).
    struct Job: Codable, Hashable, Sendable, Identifiable {
        /// `plex-sync`, `jellyfin-sync`, `arr-sync`, `disk-space-snapshot`.
        let id: String
        let name: String
        /// "Every hour", "Daily at 3:00 AM".
        let schedule: String
        let description: String

        /// The Can't Find Check (0.46+): its row also sets the wait.
        static let notFoundCheckID = "not-found-check"
    }

    /// `GET`/`PUT /settings/not-found` (admin, 0.46+): the Can't Find
    /// Check's wait, in hours after approval.
    struct NotFoundSettings: Codable, Hashable, Sendable {
        var afterHours: Int

        static let defaultAfterHours = 24
        /// What the server takes: a whole number of hours, 1 to 720 (30 days).
        static let allowedHours = 1...720
    }
}

File: windows/Marquee.Core/Models/JobsModels.cs
namespace Marquee.Core.Models;

// Settings, Jobs (api-v1.md section 13). Mirrors
// mac/Marquee/API/Models/JobsModels.swift.

/// <summary>A scheduled maintenance job ("Run now" is <c>Jobs.RunAsync(job.Id)</c>).</summary>
public sealed record Job
{
    /// <summary>
    /// <c>plex-sync</c>, <c>jellyfin-sync</c>, <c>arr-sync</c>,
    /// <c>plex-watchlist</c>, <c>disk-space-snapshot</c>, <c>cleanup</c>. Open: a job a newer server
    /// adds still lists and runs, the app just can't name it.
    /// </summary>
    public required JobId Id { get; init; }

    public required string Name { get; init; }

    /// <summary>"Every hour", "Daily at 3:00 AM".</summary>
    public required string Schedule { get; init; }

    public required string Description { get; init; }
}
